presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClient.java

```diff
@@ -0,0 +1,81 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.oracle;
+
+import io.prestosql.plugin.jdbc.BaseJdbcClient;
+import io.prestosql.plugin.jdbc.BaseJdbcConfig;
+import io.prestosql.plugin.jdbc.ColumnMapping;
+import io.prestosql.plugin.jdbc.ConnectionFactory;
+import io.prestosql.plugin.jdbc.JdbcTypeHandle;
+import io.prestosql.plugin.jdbc.StatsCollecting;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.type.VarcharType;
+
+import javax.inject.Inject;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.Optional;
+
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.decimalColumnMapping;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.varcharColumnMapping;
+import static io.prestosql.spi.type.DecimalType.createDecimalType;
+import static io.prestosql.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.prestosql.spi.type.VarcharType.createVarcharType;
+import static java.lang.Math.max;
+
+public class OracleClient
+        extends BaseJdbcClient
+{
+    @Inject
+    public OracleClient(BaseJdbcConfig config, @StatsCollecting ConnectionFactory connectionFactory)
+    {
+        super(config, "\"", connectionFactory);
+    }
+
+    @Override
+    protected ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        // Here we put TABLE and SYNONYM when the table schema is another user schema
+        return connection.getMetaData()
+                .getTables(null,
+                            schemaName.orElse(null),
+                            tableName.orElse(null),
+                            new String[] {"TABLE", "SYNONYM", "VIEW"});
+    }
+
+    @Override
+    public Optional<ColumnMapping> toPrestoType(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        int columnSize = typeHandle.getColumnSize();
+        switch (typeHandle.getJdbcType()) {
+            case Types.CLOB:
+                return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
+            case Types.NUMERIC:
+                int precision = columnSize == 0 ? 38 : columnSize;
+                int scale = max(typeHandle.getDecimalDigits(), 0);
+                return Optional.of(decimalColumnMapping(createDecimalType(precision, scale)));
+            case Types.LONGVARCHAR:
+                if (columnSize > VarcharType.MAX_LENGTH || columnSize == 0) {
+                    return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
+                }
+                return Optional.of(varcharColumnMapping(createVarcharType(columnSize)));
+            default:
```

Comment:
Nit: we can remove "default" and put the return after the switch statement (on the outside)

Reply:
done